AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -23,9 +31,112 @@ class DoubleVideoViewController: UIViewController, CellTitled {
 
     override func viewDidLayoutSubviews() {
         // update your layers' frames here
+        if let sublayers = self.videoContainerTop.layer.sublayers {
+            for layer in sublayers {
+                layer.frame = self.videoContainerTop.bounds
+            }
+        }
+        
+        if let sublayers = self.videoContainerBottom.layer.sublayers {
+            for layer in sublayers {
+                layer.frame = self.videoContainerBottom.bounds
+            }
+        }
     }
     
     @IBAction func loadVideo(_ sender: UIButton) {
+        showImagePickerForSourceType(sourceType: .photoLibrary, fromButton: sender)
+    }
+    
+    private func showImagePickerForSourceType(sourceType: UIImagePickerControllerSourceType, fromButton button: UIButton) {
+        
+        let imagePickerController = UIImagePickerController()
+        imagePickerController.modalPresentationStyle = .currentContext
+        imagePickerController.sourceType = sourceType
+        imagePickerController.delegate = self
+        imagePickerController.modalPresentationStyle = .popover
+        
+        // 3. Allow choice of images and movies
+        //    Image only is the default
+        imagePickerController.mediaTypes = [String(kUTTypeMovie)]
+        
+        self.present(imagePickerController, animated: true, completion: nil)
+    }
+    
+    func addPlayerToView(myView: UIView, item: AVPlayerItem){
+        let player = AVPlayer(playerItem: item)
+        let playerLayer = AVPlayerLayer(player: player)
+        
+        switch myView {
+        case videoContainerTop:
+            self.playerTop = player
+        case videoContainerBottom:
+            self.playerBottom = player
+        default:
+            print("unknow view")
+        }
+        
+        playerLayer.frame = myView.bounds
+        
+        let _ = self.videoContainerTop.layer.sublayers?.map{ $0.removeFromSuperlayer() }
+        myView.layer.addSublayer(playerLayer)
+        playerLayer.player?.playImmediately(atRate: 1.0)
+    }
+    
+    private func finishAndUpdate() {
+        self.dismiss(animated: true) {
+            if let url = self.videoURL.last {
+                
+                let playerItem = AVPlayerItem(url: url)
+                
+                if self.playerTop == nil{
+                    self.addPlayerToView(myView: self.videoContainerTop, item: playerItem)
+                    return
+                    
+                }
+                
+                if self.playerBottom == nil{
+                    self.addPlayerToView(myView: self.videoContainerBottom, item: playerItem)
+                    return
+                }
+                
+                if let item = self.playerTop?.currentItem{
+                    if item.currentTime().seconds / item.duration.seconds == 1{
```

Comment:
I see you're detecting the end here but the ```rate``` property of the player was specified and easier. Generally, it's very tricky checking for equality between ```Double```s because of how they're stored as very close approximations.